internal/vdso.go

```diff
@@ -16,6 +17,8 @@ var (
 	errAuxvNoVDSO = errors.New("no vdso address found in auxv")
 )
 
+const uintptrIs32bits = unsafe.Sizeof((uintptr)(0)) == 4
```

Comment:
Is the size of `uintptr` known at build time? Is this something we can fix with build constraints?

Reply:
We'd have to fix it by maintaining a list of build tags like we do here: https://github.com/cilium/ebpf/blob/main/internal/sys/ptr_64.go https://github.com/cilium/ebpf/blob/main/internal/sys/ptr_32_be.go

Reply:
Is that something you'd want to do, or do you prefer this existing way?

Reply:
I thought about this a bit, and I think the current approach is easier to maintain. With build tags there is always a part of the tree that isn't covered by regular dev environment. Any mistakes only blow up in CI.

The constant might look kind of hacky but AFAIK the compiler will eliminate the dead code. So we get the same speed with better dev experience.

Reply:
This is fine. :wink: